### PR TITLE
azure_rm_virtualmachine: support user assigned identities

### PR DESCRIPTION
**Pull request by joshsammut**
##### SUMMARY
Virtual machines can be provisioned and assigned a system assigned identity. This pull request adds support to also allow assigning a number of user assigned identities or both a system identity and user assigned identities.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
azure_rm_virtualmachine

Example useage:
```
- name: Create VM
  azure_rm_virtualmachine:
    resource_group: "group
    name: "new_vm"
    vm_size: "Standard_DS1_v2"
    network_interfaces: "{{nic}}"
    image: "{{ image }}"
    vm_identity: "UserAssigned"
    vm_user_assigned_identities: ["new_identity", "other_identity"]
```

### CODE REVIEW

**Comment by Fred-sun** [comment]
@joshsammut Thanks for your contribution, When you finished changed, I will push to review! Thanks!

**Comment by Fred-sun** [comment]
@joshsammut Thanks for your contribution, Could you can solve the ci fail? Thank you very much!

**Comment by joshsammut** [comment]
@Fred-sun sorry, had a busy week last week. I'm new to contributing to ansible, I understand the failures but was wondering if there's a way to running the integration tests locally or can they only be done through shippable?

**Comment by Fred-sun** [comment]
@joshsammut Thanks for your update! Maybe you can try this [link](https://docs.ansible.com/ansible/latest/dev_guide/testing_integration.html?highlight=integration)! Thanks!

File: lib/ansible/modules/cloud/azure/azure_rm_virtualmachine.py [hunk 1]
```diff
@@ -797,7 +805,8 @@ def __init__(self):
             zones=dict(type='list'),
             accept_terms=dict(type='bool', default=False),
             license_type=dict(type='str', choices=['Windows_Server', 'Windows_Client']),
-            vm_identity=dict(type='str', choices=['SystemAssigned']),
+            vm_identity=dict(type='str', choices=['SystemAssigned', 'UserAssigned', 'SystemAssigned, UserAssigned']),
+            vm_user_assigned_identities=dict(type='list', default=None),
```
Comment (yuwzho) [inline lib/ansible/modules/cloud/azure/azure_rm_virtualmachine.py]: Default to None can be omitted

File: lib/ansible/modules/cloud/azure/azure_rm_virtualmachine.py [hunk 2]
```diff
@@ -1170,7 +1180,21 @@ def exec_module(self, **kwargs):
                         vm_resource.license_type = self.license_type
 
                     if self.vm_identity:
-                        vm_resource.identity = self.compute_models.VirtualMachineIdentity(type=self.vm_identity)
+                        identity_args = {
```
Comment (yuwzho) [inline lib/ansible/modules/cloud/azure/azure_rm_virtualmachine.py:1319]: Can we update this?

File: lib/ansible/modules/cloud/azure/azure_rm_virtualmachine.py [hunk 3]
```diff
@@ -326,10 +326,18 @@
...
+              will assign a user assigned identity and 'SystemAssigned, UserAssigned' will assign both
         version_added: 2.8
         choices:
-            - SystemAssigned
+            - 'SystemAssigned'
+            - 'UserAssigned'
+            - 'SystemAssigned, UserAssigned'
+    vm_user_assigned_identities:
+        description:
+            - List of user assigned identity names to assign to the VM. Only useful with vm_identity='UserAssigned' or 'SystemAssigned, UserAssigned'.
```
Comment (yuwzho) [inline lib/ansible/modules/cloud/azure/azure_rm_virtualmachine.py]: Only used when I(vm_identity=UserAssigned) or I(vm_identity='SystemAssigned, UserAssigned').

**Comment by Fred-sun** [comment]
@joshsammut Could you can resolve the conflicting files and changed according by the above comment? Thank a lot!

File: lib/ansible/modules/cloud/azure/azure_rm_virtualmachine.py (hunk 2 above)
Comment (joshsammut) [reply lib/ansible/modules/cloud/azure/azure_rm_virtualmachine.py:1319]: Sorry, not sure what you mean

**Comment by Fred-sun** [comment]
@joshsammut Please help fix the shippable error when you're free! Thank you very much!

**Comment by Fred-sun** [comment]
@yungezz @zikalino @yuwzho reiview_need

**Comment by Fred-sun** [comment]
@joshdrake When you are free, please help to rebase PR and promote the merger of PR! Thanks a lot!

**Comment by Fred-sun** [comment]
@joshsammut Are you still here? Please help resolve the conflicts and promote the PR to review! Thank you very much!

**Comment by Fred-sun** [comment]
kindly ping

**Comment by Fred-sun** [comment]
@joshsammut CI fail (timeout), please retrigger CI check(close and reopen it)  and push for review! Thanks a lot!

**Comment by Fred-sun** [comment]
ready_for_review

**Comment by Fred-sun** [comment]
ping to review!

**Comment by Fred-sun** [comment]
@mybayern1974 @haiyuazhang review_need

**Comment by Fred-sun** [comment]
@gavinfish @haiyuazhang review_need!

**Comment by Fred-sun** [comment]
@gavinfish @haiyuazhang review_need!

**Comment by Fred-sun** [comment]
@joshsammut Thank you for taking the time to contribute to this PR. We will transfer ansible 's azure module related Issue and PR to azure collection (https://github.com/ansible-collections/azure/pulls), can you transfer the Issue to azure collection repo?

**Comment by mkrizek** [comment]
!component =lib/ansible/modules/cloud/azure/azure_rm_virtualmachine.py

**Comment by mkrizek** [comment]
Thank you very much for your interest in Ansible. This plugin is no longer maintained in this repository and has been migrated to https://github.com/ansible-collections/azure.

Please re-submit this issue in the above repository.

If you have further questions please stop by IRC or the mailing list:

* IRC: #ansible on irc.freenode.net
* mailing list: https://groups.google.com/forum/#!forum/ansible-project

**Pull request closed without merging**